client/src/js/components/bhPDFPrint.js

```diff
@@ -17,13 +17,14 @@ angular.module('bhima.components')
 .component('bhPdfLink', {
   bindings : {
     pdfUrl : '@',
+    buttonText : '@',
     disableCache: '@',
     options : '<',
   },
   transclude : true,
-  template :
+  template : 
     '<a href ng-click="$ctrl.print()">' +
-      '<span ng-if="!$ctrl.$loading"><i class="fa fa-print"></i> {{ "FORM.BUTTONS.PRINT" | translate }}</span>' +
+      '<span ng-if="!$ctrl.$loading"><i class="fa fa-print"></i> {{ $ctrl.buttonText | translate }}</span>' +
```

Comment:
What happened if the `buttonText` it is not set ? I think you must set a default value for `buttonText`.